include/bluetooth/audio/hap.h

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright (c) 2021 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_SERVICES_HAP_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_SERVICES_HAP_H_
+
+/**
+ * @brief Hearing Access Profile (HAP)
+ *
+ * @defgroup bt_hap Hearing Access Profile (HAP)
+ *
+ * @ingroup bluetooth
+ * @{
+ *
+ * The Hearing Access Profile is used to identify a hearing aid and optionally
+ * to control hearing aid presets.
+ *
+ * [Experimental] Users should note that the APIs can change as a part of
+ * ongoing development.
+ */
+
+#include <zephyr/types.h>
+#include <bluetooth/audio/has.h>
+#include <bluetooth/bluetooth.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Opaque Hearing Access Profile object. */
+struct bt_hap;
+
+struct bt_hap_cb {
+	/**
+	 * @brief A profile has been connected.
+	 *
+	 * This callback notifies the application of a new profile connection.
+	 * In case the err parameter is non-zero it means that the profile
+	 * connection failed.
+	 *
+	 * @param hap Profile instance object if any.
+	 * @param ha_type Hearing Aid Type.
+	 * @param error Zero for success, non-zero otherwise.
+	 */
+	void (*connected)(struct bt_hap *hap, uint8_t ha_type, int error);
+
+	/**
+	 * @brief A profile has been disconnected.
+	 *
+	 * This callback notifies the application that a profile
+	 * has been disconnected.
+	 *
+	 * @param hap Profile instance object.
+	 * @param reason Reason for the disconnection.
+	 */
+	void (*disconnected)(struct bt_hap *hap, int reason);
```

Comment:
What does `connected` and `disconnected` really mean in this context?

I assume `connected` is called when the discovery procedure is done, but I don't understand the `disconnected`. 

Can you elaborate?